gradio/exceptions.py

```diff
@@ -40,3 +43,51 @@ def __init__(self, message: str):
 
     def __str__(self):
         return repr(self.message)
+
+
+def log_message(message: str, level: Literal["info", "warning"] = "info"):
+    from gradio import queueing
+
+    if not hasattr(queueing.thread_data, "blocks"):  # Function called outside of Gradio
+        if level == "info":
+            print(message)
+        elif level == "warning":
+            warnings.warn(message)
+        return
+    if not queueing.thread_data.blocks.enable_queue:
+        warnings.warn(
+            f"Queueing must be enabled to issue {level.capitalize()}: '{message}'."
+        )
+        return
+    print("here")
+    queueing.thread_data.blocks._queue.log_message(
+        event_id=queueing.thread_data.event_id, log=message, level=level
+    )
+
+
+@document()
```

Comment:
nit but these aren't exceptions so I think it'd be better if they were in a different module